lib/animina_web/components/select_flags_component.ex

```diff
@@ -79,24 +79,28 @@ defmodule AniminaWeb.SelectFlagsComponent do
             aria-label="button"
             phx-click={
               if(
-                get_cursor_styling(
+                get_flag_styling(
                   @can_select,
                   @selected_flags,
                   flag.id,
-                  @opposite_color_flags_selected
+                  @opposite_color_flags_selected,
+                  flag,
+                  @color
                 ) == "cursor-pointer",
                 do: "select_flag",
                 else: nil
               )
             }
-            class={"rounded-full flex gap-2 items-center  px-3 py-1.5 text-sm font-semibold leading-6  focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2  #{get_cursor_styling(
+            class={"rounded-full flex gap-2 items-center  px-3 py-1.5 text-sm font-semibold leading-6  focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2  #{get_flag_styling(
               @can_select,
               @selected_flags,
               flag.id,
-              @opposite_color_flags_selected
+              @opposite_color_flags_selected,
+              flag,
```

Comment:
Do we need `flag` and `flag.id`? Wouldn't be `flag` enough?

Reply:
Here "flag_id" is for the record from the "flag_traits" table while flag.id is from the "user_flags" table

Reply:
Puhhh... a bit misleading. But we have bigger fish to fry right now.